frame/support/src/traits/tokens/fungibles.rs

```diff
@@ -281,9 +275,41 @@ pub trait Destroy<AccountId>: Inspect<AccountId> {
 	///
 	/// If successful, this function will return the actual witness data from the destroyed asset.
 	/// This may be different than the witness data provided, and can be used to refund weight.
-	fn destroy(
-		id: Self::AssetId,
-		witness: Self::DestroyWitness,
-		maybe_check_owner: Option<AccountId>,
-	) -> Result<Self::DestroyWitness, DispatchError>;
+	fn start_destroy(id: Self::AssetId, maybe_check_owner: Option<AccountId>) -> DispatchResult;
+
+	/// Destroy all accounts associated with a given asset.
+	/// `destroy_accounts` should only be called after `start_destroy` has been called, and the
+	/// asset is in a `Destroying` state
+	///
+	/// Due to weight restrictions, this function may need to be called multiple
+	/// times to fully destroy all accounts. It will destroy `RemoveItemsLimit` accounts at a
+	/// time.
+	///
+	/// - `id`: The identifier of the asset to be destroyed. This must identify an existing asset.
+	///
+	/// Each call Emits the `Event::DestroyedAccounts` event.
+	fn destroy_accounts(id: Self::AssetId, max_items: u32) -> Result<u32, DispatchError>;
```

Comment:
left wondering what the u32 returned is. Is it the amount left or destroyed? Be good to be explicit about what it's returning.

Reply:
That's true. I'll add it to the comment.